From 5171f7ca5c5e9b7bbcbfd40a16e97247f142e318 Mon Sep 17 00:00:00 2001
From: Fugang Duan <b38611@freescale.com>
Date: Mon, 11 Jan 2016 16:54:30 +0800
Subject: [PATCH 0827/5242] MLK-12184 drm/imx: imx-ldb: add mux id check

commit  02275395cca61b25ed4a61a4b087b23e3fb860c5 from
https://source.codeaurora.org/external/imx/linux-imx.git

Add mux id check and transfer a valid mux to .imx_ldb_set_clock().

Signed-off-by: Fugang Duan <B38611@freescale.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/drm/imx/imx-ldb.c |   11 +++++++++--
 1 file changed, 9 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/imx/imx-ldb.c b/drivers/gpu/drm/imx/imx-ldb.c
index 3f02ee5..3be4b47 100644
--- a/drivers/gpu/drm/imx/imx-ldb.c
+++ b/drivers/gpu/drm/imx/imx-ldb.c
@@ -241,8 +241,9 @@ static void imx_ldb_encoder_enable(struct drm_encoder *encoder)
 		else if (imx_ldb_ch == &ldb->channel[1])
 			lvds_mux = &ldb->lvds_mux[1];
 
-		regmap_update_bits(ldb->regmap, lvds_mux->reg, lvds_mux->mask,
-				   mux << lvds_mux->shift);
+		if (lvds_mux)
+			regmap_update_bits(ldb->regmap, lvds_mux->reg, lvds_mux->mask,
+					   mux << lvds_mux->shift);
 	}
 
 	regmap_write(ldb->regmap, IOMUXC_GPR2, ldb->ldb_ctrl);
@@ -264,6 +265,12 @@ static void imx_ldb_encoder_enable(struct drm_encoder *encoder)
 	int mux = drm_of_encoder_active_port_id(imx_ldb_ch->child, encoder);
 	u32 bus_format = imx_ldb_ch->bus_format;
 
+	if (mux < 0) {
+		dev_warn(ldb->dev,
+			 "%s: cannot get valid mux id\n", __func__);
+		return;
+	}
+
 	if (mode->clock > 170000) {
 		dev_warn(ldb->dev,
 			 "%s: mode exceeds 170 MHz pixel clock\n", __func__);
-- 
1.7.9.5

